geniza/corpus/metadata_export.py

```diff
@@ -189,3 +189,85 @@ class PublicDocumentExporter(DocumentExporter):
 
     def get_queryset(self):
         return super().get_queryset().filter(status=Document.PUBLIC)
+
+
+class FragmentExporter(Exporter):
+    """
+    A subclass of :class:`geniza.common.metadata_export.Exporter` that
+    exports information relating to :class:`~geniza.corpus.models.Fragment`.
+    """
+
+    model = Fragment
+    csv_fields = [
+        "shelfmark",
+        "pgpids",
+        "old_shelfmarks",
+        "collection",
+        "library",
+        "library_abbrev",
+        "collection_name",
+        "collection_abbrev",
+        "url",
+        "iiif_url",
+        "is_multifragment",
+        "created",
+        "last_modified",
+    ]
+
+    def get_queryset(self):
+        """
+        Applies some prefetching to the base Exporter's get_queryset functionality.
+
+        :return: Custom-given query set or query set of all documents
+        :rtype: QuerySet
+        """
+        return (
+            super()
+            .get_queryset()
+            .select_related("collection")
+            .prefetch_related("documents")
+        )
+
+    def get_export_data_dict(self, fragment):
+        data = {
+            "shelfmark": fragment.shelfmark,
+            "pgpids": [doc.pk for doc in fragment.documents.all()],
+            "old_shelfmarks": fragment.old_shelfmarks,
+            "url": fragment.url,
+            "iiif_url": fragment.iiif_url,
+            "is_multifragment": fragment.is_multifragment,
+            "created": fragment.created,
+            "last_modified": fragment.last_modified,
+        }
+        # it's possible (although unlikely) for collection to be unset
+        if fragment.collection:
+            # NOTE: this results in a lot of redundant info;
+            # maybe collection shortname is enough for fragment csv?
+            data.update(
+                {
+                    "collection": fragment.collection,
+                    "library": fragment.collection.library,
+                    "library_abbrev": fragment.collection.lib_abbrev,
+                    "collection_name": fragment.collection.name,
+                    "collection_abbrev": fragment.collection.abbrev,
+                }
+            )
+        return data
+
+
+# NOTE: may want a public version of fragment exporter
+# that would limit to fragments associated with public / non-suppressed documents
```

Comment:
Should we make this a separate PR or issue? Sounds easy enough, could do it via modifying get_queryset ?

Reply:
Yeah, if we go that route it would be similar to documents, where we'd have a public variant and limit it in the `get_queryset`.  I'm ambivalent about whether it's part of this PR or a separate one, what do you think?

Reply:
Maybe we ought to just tuck it into this one because it's a fairly small change and the Public/Admin is a distinction we've introduced already into the Exporter family and probably all future Exporters should have them.

Reply:
added a public version if you want to take another quick look before I merge

Reply:
Looks great!